src/zjs_callbacks.c

```diff
@@ -315,6 +334,20 @@ void zjs_remove_callback(int32_t id)
     }
 }
 
+void zjs_signal_callback_args(int32_t id, void* args, uint32_t size)
+{
+    if (args && size) {
+        int ret = zjs_port_ring_buf_put(&ring_buffer, (uint16_t)id, 0, (uint32_t*)args, (uint8_t)size / 4);
```

Comment:
ah, yes your right. I misread the implementation and did not see the line setting size32 to length:
```
if (header->length > *size32) {
    *size32 = header->length;
    return -EMSGSIZE;
}
```
So yes, instead of pulling off the queue, we can just do a while loop over the ring buffer get function and set the size to zero. If the size comes back as > 0 then there is an item in the buffer and we can pull off of it, completely doing away with the queue and solving the duplicate argument problem. Sorry for the confusion.